packages/shared/src/components/fields/ImageInput.tsx

```diff
@@ -102,6 +103,32 @@ function ImageInput({
     onChange?.(base64, file);
   };
 
+  const onDragOver = (event: DragEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+  };
+
+  const onDrop = async (event: DragEvent<HTMLButtonElement>) => {
```

Comment:
Seems like both onFileChange and onDrop do almost the same things after they have the file. Would be nice to unify these functions rather then repeat code.

Reply:
Maybe introduce a function named `validateFileInput` or something to contain the logic.